AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/SQLStatement+Select.swift

```diff
@@ -101,12 +72,81 @@ struct SelectStatement: SQLStatement {
             \(paginationInput.sqlStatement)
             """
         }
+        return SelectStatementMetadata(statement: sql,
+                                       columnMapping: columnMapping,
+                                       bindings: bindings)
+    }
+
+    /// Tuple that represents a pair of `columns` and `statements`
+    private typealias JoinStatements = ([String], [String], ColumnMapping)
+
+    /// Walk through the associations recursively to generate join statements.
+    ///
+    /// Implementation note: this should be revisited once we define support
+    /// for explicit `eager` vs `lazy` associations.
+    private static func joins(from schema: ModelSchema) -> JoinStatements {
```

Comment:
how about `generate(from: foreignKeys: [ModelField] -> JoinStatement?` ? JoinStatement in this case would be a struct like
```
struct JoinStatement {
        let columns: [String]
        let statements: [String]
        let columnMapping: ColumnMapping
    }
```
and would return nil if there were no foreign keys to generate a JoinStatement from

Reply:
done